libraries/Microsoft.Bot.Configuration/ValidationUtilities.cs

```diff
@@ -0,0 +1,151 @@
+            {
+                // If there's no file, throws an Exception
+                throw new FileNotFoundException("There isn't any \'.csproj\' in the specified folder", ex);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the <see cref="LuisService"/> has its key defined.
+        /// </summary>
+        /// <param name="botConfiguration">BotConfiguration object to extend.</param>
+        /// <returns>Boolean that indicates if a bot configuration file's Luis keys are present.</returns>
+        public static bool ValidateLuisKey(this BotConfiguration botConfiguration)
+        {
+            try
+            {
+                // Get all the luis types in the service list
+                var luisServices = botConfiguration.FindServicesByType(ServiceTypes.Luis);
+
+                // If there isn't any Luis service, throws exception
+                if (luisServices == null)
+                {
+                    throw new Exception("There is no LUIS service in the bot configuration file.");
+                }
+
+                // Check if exists any LuisService without its key (SubscriptionKey)
+                var existEmptyLuis = luisServices.Any(x => string.IsNullOrWhiteSpace(((LuisService)x).SubscriptionKey));
+
+                // If there is at least one LuisService without its key, the process will return false
+                if (existEmptyLuis)
+                {
+                    return false;
+                }
+
+                // Get all the Dispatch (Luis) types in the service list
+                var dispatchServices = botConfiguration.FindServicesByType(ServiceTypes.Dispatch);
+
+                // The dispatch service is not always included in the the bot configuration files.
+                // Therefore, it will only be validated if the bot configuration file contains at least one service of this type.
+                if (dispatchServices != null && dispatchServices.Count() != 0)
+                {
+                    // Check if exists any dispatchServices without its key (SubscriptionKey)
+                    var existEmptyDispatch = dispatchServices.Any(x => string.IsNullOrWhiteSpace(((DispatchService)x).SubscriptionKey));
+
+                    // If there is at least one dispatch service without its key, the process will return false
+                    return !existEmptyDispatch;
+                }
+
+                return true;
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Checks if the <see cref="QnAMakerService"/> has its key defined.
+        /// </summary>
+        /// <param name="botConfiguration">BotConfiguration object to extend.</param>
+        /// <returns>Boolean that indicates if a bot configuration file's QnAMaker keys are present.</returns>
+        public static bool ValidateQnAKey(this BotConfiguration botConfiguration)
```

Comment:
Same comment on name. As it's named, I would think this does a Deep Validation (hitting QnA, etc). 

This should just be "ValidateQnAKeyPresent" or similar.

Reply:
Also, same comment on the return type. Consider a Tuple (bool, string) where the string is the failure reason.

Reply:
Now, the method name is `ValidateQnAKeyExistence`. Also, we implemented a tuple which contains a boolean and a string (error message).